docs/cicd_notes.md

```diff
@@ -1,7 +1,84 @@
+The developed action will build and push to Docker Hub the test container every time the files it depends on have changed on the main branch. Also, the pushed image will have the **main** tag in Docker Hub.
 
 As a note, we created an [Access Token](https://docs.docker.com/docker-hub/access-tokens/) in Docker Hub to use instead of the account's password. This way, Docker Hub lets users monitor what was done using that token, and helps avoid security issues.
+
+## Choosing a CI platform
+We had to choose two CI platforms to start automatically executing the unit tests when trying to make any changes to the repository's code. To choose the platforms we used the following criteria:
+
+1. It must have a permanently free version that gives a certain number of credits to open source projects that do not force the user to add billing information.
+2. The free credits must be monthly or yearly renewed and the number of credits per month must be as high as possible.
+3. It must be easy to integrate with GitHub.
+4. It must be easy to install and configure.
+5. The platform must be hosted by a third party since we don't have the infrastructure necessary for self-hosting -sorry Jenkins-.
+6. It must not have frequent availability problems.
+
+The options considered are the following:
+
+### GitHub Actions
+#### Description
+A tool developed by GitHub to automate software workflows. It allows developers and organizations not only to perform the typical CI/CD workflows but also to manage their repositories.
+
+#### Analysis
+- The use of GitHub Actions is totally free for public repositories so the first two points are covered.
+- It's really easy to integrate with GitHub since it's designed by the GitHub team.
+- Configuring a GitHub Action is very easy, the developer just needs to upload a YAML file with all the steps to the repository under the `.github/workflows` folder. We have already used a GitHub action to build the container we use to pass the unit tests of the project, so we are already familiarized with its configuration and usage.
+- It's hosted by GitHub.
+- It had not any recent availability problems that we could notice.
+
+#### Conclusion
+GitHub Actions got 6 out of 6 points.
+
+### Semaphore CI
+#### Description
+A CI/CD tool that [advertises itself](https://semaphoreci.com/product) as a fast tool. It has a free tier and it has good reviews on sites like [g2.com](https://www.g2.com/products/semaphore/reviews). 
+
+#### Analysis
+- They have a free tier.
+- They give the equivalent of $10 in credits per month to use for free, around 1300 minutes.
+- The platform is Cloud-based.
+
+We didn't check the rest of the points since we faced different errors with the service the whole day we tried to use it.
```

Comment:
You should have documented this; I will take your word for it, but really the objective script requested all these things to be documented.

Reply:
I had documented it but I lost the screenshots and I wasn't able to take them again after having chosen AppVeyor :facepalm: